backstory/src/test/java/com/google/sps/story/DatamuseRequestClientTest.java

```diff
@@ -0,0 +1,337 @@
+    
+      Assert.assertEquals(EXPECTED_MESSAGE, exception.getMessage());
+      Assert.assertEquals(JSONException.class, exception.getCause().getClass());
+      return; // this is the behavior we wanted to happen
+    }
+
+    Assert.fail("No RuntimeException was thrown when parsing incorrect JSON.");
+  }   
+
+  /** 
+   * Check that fetchRelatedAdjectives() throws a RunTimeException
+   * when the JSON array that it gets back from the server does not
+   * contain JSON objects.
+   */
+  @Test 
+  public void jsonArrayButNotJsonObjects() throws Exception {
+    final String NOUN = "beach";
+    final int MAX_ADJECTIVES = 5;
+    final String QUERIES = "rel_jjb=" + NOUN + "&max=" + MAX_ADJECTIVES;
+    final String[] INPUT = { "sandy", "long", "private", "white", "small"}; 
+
+    DatamuseRequestClient client = new DatamuseRequestClient(DEFAULT_URL);
+
+    // inject the url but with a response that's an array but of Strings not of JSON objects
+    injectURLConnectionMock(DEFAULT_URL, QUERIES, Arrays.toString(INPUT));
+
+    try {
+      String[] actualOutput = client.fetchRelatedAdjectives(NOUN, MAX_ADJECTIVES);
+    } catch (RuntimeException exception) {
+      final String EXPECTED_MESSAGE = "JSON array received was not of JSON objects.";
+    
+      Assert.assertEquals(EXPECTED_MESSAGE, exception.getMessage());
+      Assert.assertEquals(ClassCastException.class, exception.getCause().getClass());
+      return; // this is the behavior we wanted to happen
+    }
+
+    Assert.fail("No RuntimeException was thrown when objects were not valid JSON.");
+  }
+
+  /** 
+   * Check that fetchRelatedAdjectives() throws a RuntimeException
+   * when the JSON objects in the JSON array do not have a String
+   * value for "word".
+   */
+  @Test 
+  public void noValueForWord() throws Exception {
+    final String NOUN = "beach";
+    final int MAX_ADJECTIVES = 5;
+    final String QUERIES = "rel_jjb=" + NOUN + "&max=" + MAX_ADJECTIVES;
+    
+    JSONArray input = new JSONArray();
+    input.put(new JSONObject()); // put in a default object without a "word" field
+
+    DatamuseRequestClient client = new DatamuseRequestClient(DEFAULT_URL);
+
+    // inject the url but with a response that doesn't have a "word" field
+    injectURLConnectionMock(DEFAULT_URL, QUERIES, input.toString());
+
+    try {
+      String[] actualOutput = client.fetchRelatedAdjectives(NOUN, MAX_ADJECTIVES);
+    } catch (RuntimeException exception) {
+      final String EXPECTED_MESSAGE = "JSON Object did not have a String value for the key \"word\".";
+    
+      Assert.assertEquals(EXPECTED_MESSAGE, exception.getMessage());
+      Assert.assertEquals(JSONException.class, exception.getCause().getClass());
+      return; // this is the behavior we wanted to happen
+    }
+
+    Assert.fail("No RuntimeException was thrown when parsing incorrect JSON.");
+  }   
+
+  /** 
+   * Check that fetchRelatedAdjectives() works for when less
```

Comment:
"works for" -> "succeeds"?

Reply:
That sounds better. I'll make the change.

Reply:
Done